File: Array/C++/BuyandSellStock.cpp
#include <bits/stdc++.h>
#define log(x) cout<<x<<endl;
using namespace std;

void print(vector<int> a){
    for(auto i: a){
        cout<<i<<" ";
    }
    cout<<endl;
}

int main()
{
    vector<int> a = {7,5,3,6,4,1}; 
    int profit =0, max=0;
    for(int i=1; i<a.size(); i++){
        profit += a[i] - a[i-1];
        if(max<profit) max = profit;
        if(profit<0) profit = 0;
    }
    log(max)
}

File: Array/Python/BuyandSellStock.py
a = [2,4,1]

profit, max = 0,0
for i in range(1, len(a)):
    profit += a[i] - a[i-1]
    if(profit > max):
        max = profit
    if(profit < 0):
        profit = 0

print(max)
